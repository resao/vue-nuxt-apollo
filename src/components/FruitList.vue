<template>
  <div class="all-fruits">
    <h2>All Fruits</h2>
    <ul v-if="allFruits">
      <li v-for="fruit in allFruits" :key="fruit.id">{{fruit.name}} <em v-if="fruit.color">{{fruit.color.name}}</em></li>
    </ul>
  </div>
</template>

<script>
import allFruitQuery from '@/mixins/allFruitQuery'
export default {
  mixins: [allFruitQuery]
}
</script>
