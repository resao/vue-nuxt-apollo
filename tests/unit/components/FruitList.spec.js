import { shallowMount } from '@vue/test-utils'
import FruitList from '@/components/FruitList.vue'

describe('FruitList.vue', () => {
  it('should match html snapshot', () => {
    const wrapper = shallowMount(FruitList, {
      data () {
        return {
          allFruits: [{
            name: 'Apple',
            color: {
              name: 'Green'
            }
          }, {
            name: 'Banana',
            color: {
              name: 'Yellow'
            }
          }, {
            name: 'Banana',
            color: {
              name: 'Yellow'
            }
          }]
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
