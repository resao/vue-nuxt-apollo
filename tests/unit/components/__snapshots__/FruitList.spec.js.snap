// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FruitList.vue should match html snapshot 1`] = `
<div class="fruit-list">
  <h2>All Fruits</h2>
  <ul>
    <li>Apple <em>Green</em></li>
    <li>Banana <em>Yellow</em></li>
    <li>Banana <em>Yellow</em></li>
  </ul>
</div>
`;
